### ISSUE TITLE
Deployment failed - ingestEvents->runPowerShellInline

### ISSUE THREAD

**Issue by romandas** [Azure/Azure-Sentinel#4725]
**Describe the bug**
Using the instructions in the training guide, deployment of initial Azure Sentinel Training Lab failed on ingestEvents, in the runPowerShellInline script. Error provided is:
DeploymentScriptOperationFailed
"The received access token is not valid: at least one of the claims 'puid' or 'altsecid' or 'oid' should be present. If you are accessing as application please make sure service principal is properly created in the tenant."

This deployment was using a newly created Azure account using the trial free period. Deployed to East US.

**To Reproduce**
Follow the instructions listed here: https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Training/Azure-Sentinel-Training-Lab/Modules/Module-1-Setting-up-the-environment.md
Deployment failed at Exercise 2, step 4.

**Expected behavior**
Successful deployment

**Screenshots**
![Screenshot_2022-04-26_22-54-41](https://user-images.githubusercontent.com/1872679/165430370-192b186a-7e39-4d88-b093-405debdf278c.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Firefox
 - Version: 99

**Comment by azurekid** [comment]
Hi @romandas,

I have just validated this in my environment, and it seems to work as expected.
![image](https://user-images.githubusercontent.com/40334679/165592266-8d9c8d32-7e0c-42dc-acf9-2d09a7203b36.png)

When looking into the deployment method for data ingesting data deploymentScript resource is used.
Can you share the error in which is shown in the 'Operation details' of the deployment? I have seen a GatewayTimeOut before, but that was gone when I re-deployed the solution.

**Comment by romandas** [comment]
Yes, the error was:
DeploymentScriptOperationFailed
"The received access token is not valid: at least one of the claims 'puid' or 'altsecid' or 'oid' should be present. If you are accessing as application please make sure service principal is properly created in the tenant."

It failed three times -- twice yesterday, and once this morning. Though, it just worked now, on the fourth try. Not sure why. I wonder if it is related to my just creating a new Azure account?

**Comment by azurekid** [comment]
What a Deployment Script deployment does is temporarily create a _Container Instance_ and _StorageAccount_ that is used to run the deployment script in.
![image](https://user-images.githubusercontent.com/40334679/165598940-da2b2576-7f2c-44f1-807b-263375caa64a.png)

To run the script from the container a used assigned managed identity is also created.
Another thing that sometimes happens is that the user that starts the deployment has insufficient permissions to assign a role to the _user assigned managed identity_ causing the deployment to fail.

Does that help?

**Comment by romandas** [comment]
I didn't change any permissions on my account -- it's a personal one using the Azure free trial -- so I don't know if that would really be the issue in this case. So, could there be a race condition here, where the managed identity hasn't been created yet when the script runs?

**Comment by javiersoriano** [comment]
@romandas have you experienced this error again? Yeah, it might be a race condition, but we have no control over that, we just use deploymentScripts resource type and that takes care of creating the different components including the managed identity and granting permissions to it

**Comment by jackchenwork** [comment]
I got the same error today, reinstalled the training lab multiple times and it still fail.

![image](https://user-images.githubusercontent.com/28953194/210427411-214dfb19-516b-471b-b59a-d0247dd23c27.png)

I did a bit investigation, the job failed on a deployment script IngestCSV.ps1
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Training/Azure-Sentinel-Training-Lab/Artifacts/Scripts/IngestCSV.ps1

It require two parameters and I confirmed the workspaceID and shared key are correct. I see the script will create 6 custom log tables, and the deployment script only created 5, the biggested securityEvents_CL table was not created.

I tried to manually run IngestCSV.ps1 and it works fine ( 6 tables were created ). I suspect Azure deployment script might have a resource issue to handle the bigger table and somehow it failed. 

Workaround is manually run IngestCSV.ps1 with correct workspace ID and key.
